### PR TITLE
fix(s3stream): avoid MAJOR_V1 compact composite object itself

### PR DESCRIPTION
**Pull request by lifepuzzlefun**

### CODE REVIEW

**Comment by lifepuzzlefun** [comment]
log 
```shell
[2024-07-09 18:45:17,499] TRACE [StreamObjectsCompactionTask streamId=358] MAJOR_V1 CompactStreamObjectRequest{objectId=3090, objectSize=20701, streamId=358, startOffset=0, endOffset=5962510, streamEpoch=0, attributes=1, sourceObjectIds=[978, 994, 2131, 2142], operations=[KEEP_DATA, KEEP_DATA, KEEP_DATA, KEEP_DATA]} cost 297ms (s3.object.logger)
[2024-07-09 19:45:40,385] TRACE [StreamObjectsCompactionTask streamId=358] MAJOR_V1 CompactStreamObjectRequest{objectId=7087, objectSize=56987, streamId=358, startOffset=0, endOffset=20787253, streamEpoch=0, attributes=1, sourceObjectIds=[3090, 3813, 4982, 4992, 6147, 6158], operations=[DELETE, KEEP_DATA, KEEP_DATA, KEEP_DATA, KEEP_DATA, KEEP_DATA]} cost 168ms (s3.object.logger)
[2024-07-09 20:46:08,962] TRACE [StreamObjectsCompactionTask streamId=358] MAJOR_V1 CompactStreamObjectRequest{objectId=10146, objectSize=68053, streamId=358, startOffset=0, endOffset=25259981, streamEpoch=0, attributes=1, sourceObjectIds=[7087, 7677], operations=[DELETE, KEEP_DATA]} cost 127ms (s3.object.logger)
[2024-07-09 21:46:30,177] TRACE [StreamObjectsCompactionTask streamId=358] MAJOR_V1 CompactStreamObjectRequest{objectId=11588, objectSize=68053, streamId=358, startOffset=0, endOffset=25259981, streamEpoch=0, attributes=1, sourceObjectIds=[10146], operations=[DELETE]} cost 54ms (s3.object.logger)
[2024-07-09 22:46:48,171] TRACE [StreamObjectsCompactionTask streamId=358] MAJOR_V1 CompactStreamObjectRequest{objectId=13157, objectSize=68053, streamId=358, startOffset=0, endOffset=25259981, streamEpoch=0, attributes=1, sourceObjectIds=[11588], operations=[DELETE]} cost 55ms (s3.object.logger)
[2024-07-09 23:47:05,071] TRACE [StreamObjectsCompactionTask streamId=358] MAJOR_V1 CompactStreamObjectRequest{objectId=14709, objectSize=68053, streamId=358, startOffset=0, endOffset=25259981, streamEpoch=0, attributes=1, sourceObjectIds=[13157], operations=[DELETE]} cost 46ms (s3.object.logger)
```

File: s3stream/src/main/java/com/automq/stream/s3/compact/StreamObjectCompactor.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ static boolean checkObjectGroupCouldBeCompact(List<S3ObjectMetadata> objectGroup
         if (objectGroup.size() == 1 && (MINOR.equals(compactionType) || MAJOR.equals(compactionType) || MINOR_V1.equals(compactionType))) {
             return false;
         }
-        if (CLEANUP_V1.equals(compactionType)) {
+        if (CLEANUP_V1.equals(compactionType) || MAJOR_V1.equals(compactionType)) {
```
Comment (superhx) [inline s3stream/src/main/java/com/automq/stream/s3/compact/StreamObjectCompactor.java:187]: If objectsGroup has 100 objects, but the first composite object does not have dirty data, then MAJOR_V1 will not run.

Comment (superhx) [reply s3stream/src/main/java/com/automq/stream/s3/compact/StreamObjectCompactor.java:187]: The expected behavior should be: do not run MAJOR_V1 compaction if objectGroup.size() == 1.

Comment (lifepuzzlefun) [reply s3stream/src/main/java/com/automq/stream/s3/compact/StreamObjectCompactor.java:187]: got it. I'll change the code

**Review by superhx — APPROVED** [review]

**Pull request merged**